homeassistant/components/zwave/__init__.py

```diff
@@ -410,28 +409,23 @@ def set_config_parameter(service):
         node = network.nodes[node_id]
         param = service.data.get(const.ATTR_CONFIG_PARAMETER)
         selection = service.data.get(const.ATTR_CONFIG_VALUE)
-        size = service.data.get(const.ATTR_CONFIG_SIZE, 2)
-        i = 0
         for value in (
                 node.get_values(class_id=const.COMMAND_CLASS_CONFIGURATION)
                 .values()):
-            if value.index == param and value.type == const.TYPE_LIST:
-                _LOGGER.debug("Values for parameter %s: %s", param,
-                              value.data_items)
-                i = len(value.data_items) - 1
-        if i == 0:
-            node.set_config_param(param, selection, size)
-        else:
-            if selection > i:
-                _LOGGER.error("Config parameter selection does not exist! "
-                              "Please check zwcfg_[home_id].xml in "
-                              "your homeassistant config directory. "
-                              "Available selections are 0 to %s", i)
-                return
-            node.set_config_param(param, selection, size)
-            _LOGGER.info("Setting config parameter %s on Node %s "
-                         "with selection %s and size=%s", param, node_id,
-                         selection, size)
+            if value.index != param:
```

Comment:
Yes, It might be a usefull service call to have, but my opinion is:
The device should rather be added to OZW than we supporting making changes on an unknown device.